packages/x-date-pickers/src/internals/hooks/useField/useField.utils.ts

```diff
@@ -319,7 +319,7 @@ export const getSectionVisibleValue = (
 };
 
 export const cleanString = (dirtyString: string) =>
-  dirtyString.replace(/\u2066|\u2067|\u2068|\u2069/g, '');
+  dirtyString.replace(/[\u2066\u2067\u2068\u2069]/g, '');
```

Comment:
I had this suggestion on my IDE

See https://www.jetbrains.com/help/phpstorm/regexp-single-character-alternation.html